service/controller/aws/resource_set.go

```diff
@@ -406,6 +407,10 @@ func getWorkerCount(obj interface{}) (int, error) {
 	return key.WorkerCount(cr), nil
 }
 
+func getWorkerMaxCPUCores(obj interface{}) (int, bool, error) {
```

Comment:
Can you add a comment that CPU cores only applies for KVM? So it's clear why it returns 0.

Reply:
I'll try to push increase of min CPU for KVM from 2 to 4 - then could simplify/remove this max CPU cores calculation and related logic.

Reply:
Removed XS cluster profile detection based on max 2 CPU workers, assuming min will be 4 for all providers (https://github.com/giantswarm/giantswarm/issues/5063)